internal/controller/intelcluster_controller.go

```diff
@@ -200,6 +243,45 @@ func (r *IntelClusterReconciler) reconcileControlPlaneEndpoint(scope *scope.Clus
 	return true
 }
 
+func (r *IntelClusterReconciler) reconcileClusterConnectConnection(scope *scope.ClusterReconcileScope) bool {
+	intelCluster := scope.IntelCluster
+	conditions.MarkTrue(intelCluster, infrav1.ConnectionAliveCondition)
+
+	if !scope.Cluster.Status.ControlPlaneReady {
+		return false
+	}
+
+	clusterConnect := &ccgv1.ClusterConnect{}
+	if err := r.Client.Get(scope.Ctx, client.ObjectKey{
+		Name:      fmt.Sprintf("%s-%s", scope.IntelCluster.Namespace, scope.IntelCluster.Name),
+		Namespace: scope.IntelCluster.Namespace,
+	}, clusterConnect); err != nil {
+		if !apierrors.IsNotFound(err) {
+			scope.Log.Info("failed to read cluster connection resource")
+			return true
+		}
+	}
+
+	// get the connection probe condition from the clusterconnect resource
+	connectionProbeCondition := metav1.Condition{}
+	ccConditions := clusterConnect.GetConditions()
+	for _, condition := range ccConditions {
+		if condition.Type == ccgv1.ConnectionProbeCondition {
+			connectionProbeCondition = condition
+		}
+	}
+
+	if connectionProbeCondition.Status != metav1.ConditionTrue {
+		scope.Log.Info("connection probe condition not met in clusterconnect resource")
+		conditions.MarkFalse(intelCluster, infrav1.ConnectionAliveCondition, infrav1.ConnectionNotAliveReason, clusterv1.ConditionSeverityError, "No connection to cluster, waiting for connection probe condition to be true")
```

Comment:
Does setting this make IntelCluster.Status.Ready to be false?

Reply:
I've checked that and when you reviewed - it would change status to false

According to contract, the ready status shouldn't be changed after successfull provisioning

So i excluded "runtime' condition from calculating status.Ready
https://github.com/open-edge-platform/cluster-api-provider-intel/pull/43/commits/bcd0b5196a324bb09be889cfc366d319d1a3dce3